### DISCUSSION TITLE
Please help me to understand how to establish communication in my custom Lua script between radio and flight controller (Betaflight)

### DISCUSSION THREAD

**Discussion by captain-yossarian** [ExpressLRS/ExpressLRS#3066] [Q&A]
I want to practice Lua writing EdgeTX scripts for FPV drone telemetry. I have found a pretty decent documentation for [edge tx](https://luadoc.edgetx.org/) and Lua . In order to accomplish my task I need to establish a communication of my radio (transmitter) with flight controller on my drone wich has betaflight firmware [see MSP communication](https://betaflight.com/docs/development/API/MSP-Extensions). 

There is a nice lua script for communication with betaflight [see here](https://github.com/betaflight/betaflight-tx-lua-scripts) but for me it is hard to understand how it communicates with betaflight on byte level. Do you have any small easy to understand examples to dive into it ?

So far, I have found that crossfireTelemetryPush function sends a request via telemetry to receiver. This is the starting point. An example of using this function you can see here:
```lua
local function crsfDisplayPortCmd(cmd, data)
    local payloadOut = { CONST.address.betaflight, CONST.address.transmitter, cmd }
    if data ~= nil then
        for i = 1, #(data) do
            payloadOut[3 + i] = data[i]
        end
    end
    return crossfireTelemetryPush(CONST.frameType.displayPort, payloadOut) 
end
```

Another one example might be found in ELRS lua script:
```lua
crossfireTelemetryPush(0x2D, { deviceId, handsetId, fieldPopup.id, 5 })
```
[Here](https://luadoc.edgetx.org/part_iii_-_opentx_lua_api_reference/general-functions-less-than-greater-than-luadoc-begin-general/crossfiretelemetrypush) I have found a documentation to this function but as you might have noticed it is poorly explained what is going on there.

To summarize I'd like to understand more hot to make a payload and send it from radio via ELRS with help of Lua script to flight controller. Apart from that , could you pls explain me how ELRS packet is made on byte level (header, payload, ending) on lua example. HIghly appreciate it, Thank you

Before posting it I have asked a [question](https://stackoverflow.com/questions/79261586/how-to-make-a-payload-in-edge-tx-and-send-it-with-elrs-telemetry-package-to-flig) on stackoferflow and [reddit](https://www.reddit.com/r/fpv/comments/1h94k24/establish_communication_with_betaflight_with_help/)
